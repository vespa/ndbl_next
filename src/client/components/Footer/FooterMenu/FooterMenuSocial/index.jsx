
import React from 'react';
import PropTypes from 'prop-types';
import css from './style.scss';
import cssMain from '../../../../../scss/main.scss';
import FooterMenuSocialItem from './FooterMenuSocialItem';

const FooterMenuSocial = ({ title }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <div className={css['footer-menu__list']}>
        <ul className={`${cssMain.col_6} ${css['footer-menu__list']}}`}>
          {[
            { text: 'Newsletter', icon: 'newsletter', href: '/' },
            { text: 'Instagram', icon: 'instagram', href: '/' },
            { text: 'Twitter', icon: 'twitter', href: '/' },
            { text: 'Tumblr', icon: 'tumblr', href: '/' },
            { text: 'Linkedin', icon: 'linkedin', href: '/' },
          ].map(item => <FooterMenuSocialItem key={item.text} {...item} />)
          }
        </ul>
        <ul className={`${cssMain.col_6} ${css['footer-menu__list']}}`}>
          {[
            { text: 'Facebook', icon: 'facebook', href: '/' },
            { text: 'Pinterest', icon: 'pinterest', href: '/' },
            { text: 'Youtube', icon: 'youtube', href: '/' },
            { text: 'Google+', icon: 'google', href: '/' },
          ].map(item => <FooterMenuSocialItem key={item.text} {...item} />)
          }
        </ul>
      </div>
    </React.Fragment>
  );
};

FooterMenuSocial.propTypes = {
  title: PropTypes.string.isRequired,
};

export default FooterMenuSocial;
