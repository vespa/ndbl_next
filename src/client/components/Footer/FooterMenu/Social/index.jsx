
import React from 'react';
import PropTypes from 'prop-types';
import css from './style.scss';
import cssMain from '../../../../../scss/main.scss';

const FooterMenuSocial = ({ title }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <ul className={css['footer-menu__list']}>
        {/* {links.map(item => <li key={item.text}><a href={item.href}>{item.text}</a> </li>)} */}
        <ul className={cssMain.col_6}>
          teste
        </ul>
        <ul className={cssMain.col_6}>
          teste
        </ul>
      </ul>
    </React.Fragment>
  );
};

FooterMenuSocial.propTypes = {
  title: PropTypes.string.isRequired,
};

export default FooterMenuSocial;
