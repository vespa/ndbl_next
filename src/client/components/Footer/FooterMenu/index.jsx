
import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import css from './style.scss';

const FooterMenu = ({ title, links }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <ul className={css['footer-menu__list']}>
        {links.map(item => <li key={item.text}><Link href={item.href}>{item.text}</Link> </li>)}
      </ul>
    </React.Fragment>
  );
};

FooterMenu.propTypes = {
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default FooterMenu;
