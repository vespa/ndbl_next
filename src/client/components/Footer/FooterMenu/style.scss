.footer-menu {
  &__title {
    text-transform: uppercase;
  }
  &__list {
    list-style-type: none;
    padding: 0;
    margin: 0;
    a {
      color: #aaa;
      display: block;
      margin-bottom: 10px;
    }
  }
}