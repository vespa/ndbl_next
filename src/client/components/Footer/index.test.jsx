import React from 'react';
import { shallow } from 'enzyme';
import FooterMenu from './index';

describe('[Component: FooterMenu]', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FooterMenu />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
