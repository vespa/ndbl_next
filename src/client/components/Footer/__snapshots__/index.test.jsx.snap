// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: FooterMenu] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
        />,
        <div
          className={undefined}
        >
          <FooterMenu
            links={
              Array [
                Object {
                  "href": "/",
                  "text": "Home",
                },
                Object {
                  "href": "/",
                  "text": "Jobs",
                },
                Object {
                  "href": "/",
                  "text": "About",
                },
                Object {
                  "href": "/",
                  "text": "Departament",
                },
              ]
            }
            title="jobs at"
          />
        </div>,
        <div
          className={undefined}
        >
          <FooterMenu
            links={
              Array [
                Object {
                  "href": "/",
                  "text": "F.A.Q.",
                },
                Object {
                  "href": "/",
                  "text": "Application process",
                },
                Object {
                  "href": "/",
                  "text": "Privacy policy",
                },
                Object {
                  "href": "/",
                  "text": "Contact",
                },
              ]
            }
            title="Information"
          />
        </div>,
        <div
          className={undefined}
        >
          <FooterMenuSocial
            title="Folow Us"
          />
        </div>,
        <div
          className={undefined}
        >
          <FooterMenu
            links={
              Array [
                Object {
                  "href": "/",
                  "text": "Subscribe to your job alert",
                },
              ]
            }
            title="Job Alert"
          />
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FooterMenu
            links={
              Array [
                Object {
                  "href": "/",
                  "text": "Home",
                },
                Object {
                  "href": "/",
                  "text": "Jobs",
                },
                Object {
                  "href": "/",
                  "text": "About",
                },
                Object {
                  "href": "/",
                  "text": "Departament",
                },
              ]
            }
            title="jobs at"
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "links": Array [
              Object {
                "href": "/",
                "text": "Home",
              },
              Object {
                "href": "/",
                "text": "Jobs",
              },
              Object {
                "href": "/",
                "text": "About",
              },
              Object {
                "href": "/",
                "text": "Departament",
              },
            ],
            "title": "jobs at",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FooterMenu
            links={
              Array [
                Object {
                  "href": "/",
                  "text": "F.A.Q.",
                },
                Object {
                  "href": "/",
                  "text": "Application process",
                },
                Object {
                  "href": "/",
                  "text": "Privacy policy",
                },
                Object {
                  "href": "/",
                  "text": "Contact",
                },
              ]
            }
            title="Information"
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "links": Array [
              Object {
                "href": "/",
                "text": "F.A.Q.",
              },
              Object {
                "href": "/",
                "text": "Application process",
              },
              Object {
                "href": "/",
                "text": "Privacy policy",
              },
              Object {
                "href": "/",
                "text": "Contact",
              },
            ],
            "title": "Information",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FooterMenuSocial
            title="Folow Us"
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Folow Us",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FooterMenu
            links={
              Array [
                Object {
                  "href": "/",
                  "text": "Subscribe to your job alert",
                },
              ]
            }
            title="Job Alert"
          />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "links": Array [
              Object {
                "href": "/",
                "text": "Subscribe to your job alert",
              },
            ],
            "title": "Job Alert",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          />,
          <div
            className={undefined}
          >
            <FooterMenu
              links={
                Array [
                  Object {
                    "href": "/",
                    "text": "Home",
                  },
                  Object {
                    "href": "/",
                    "text": "Jobs",
                  },
                  Object {
                    "href": "/",
                    "text": "About",
                  },
                  Object {
                    "href": "/",
                    "text": "Departament",
                  },
                ]
              }
              title="jobs at"
            />
          </div>,
          <div
            className={undefined}
          >
            <FooterMenu
              links={
                Array [
                  Object {
                    "href": "/",
                    "text": "F.A.Q.",
                  },
                  Object {
                    "href": "/",
                    "text": "Application process",
                  },
                  Object {
                    "href": "/",
                    "text": "Privacy policy",
                  },
                  Object {
                    "href": "/",
                    "text": "Contact",
                  },
                ]
              }
              title="Information"
            />
          </div>,
          <div
            className={undefined}
          >
            <FooterMenuSocial
              title="Folow Us"
            />
          </div>,
          <div
            className={undefined}
          >
            <FooterMenu
              links={
                Array [
                  Object {
                    "href": "/",
                    "text": "Subscribe to your job alert",
                  },
                ]
              }
              title="Job Alert"
            />
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FooterMenu
              links={
                Array [
                  Object {
                    "href": "/",
                    "text": "Home",
                  },
                  Object {
                    "href": "/",
                    "text": "Jobs",
                  },
                  Object {
                    "href": "/",
                    "text": "About",
                  },
                  Object {
                    "href": "/",
                    "text": "Departament",
                  },
                ]
              }
              title="jobs at"
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "links": Array [
                Object {
                  "href": "/",
                  "text": "Home",
                },
                Object {
                  "href": "/",
                  "text": "Jobs",
                },
                Object {
                  "href": "/",
                  "text": "About",
                },
                Object {
                  "href": "/",
                  "text": "Departament",
                },
              ],
              "title": "jobs at",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FooterMenu
              links={
                Array [
                  Object {
                    "href": "/",
                    "text": "F.A.Q.",
                  },
                  Object {
                    "href": "/",
                    "text": "Application process",
                  },
                  Object {
                    "href": "/",
                    "text": "Privacy policy",
                  },
                  Object {
                    "href": "/",
                    "text": "Contact",
                  },
                ]
              }
              title="Information"
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "links": Array [
                Object {
                  "href": "/",
                  "text": "F.A.Q.",
                },
                Object {
                  "href": "/",
                  "text": "Application process",
                },
                Object {
                  "href": "/",
                  "text": "Privacy policy",
                },
                Object {
                  "href": "/",
                  "text": "Contact",
                },
              ],
              "title": "Information",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FooterMenuSocial
              title="Folow Us"
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Folow Us",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FooterMenu
              links={
                Array [
                  Object {
                    "href": "/",
                    "text": "Subscribe to your job alert",
                  },
                ]
              }
              title="Job Alert"
            />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "links": Array [
                Object {
                  "href": "/",
                  "text": "Subscribe to your job alert",
                },
              ],
              "title": "Job Alert",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
